src/main/java/com/ifmo/epampractice/dao/InvoiceDao.java

```diff
@@ -0,0 +1,130 @@
+            "PAYMENT_DATE, TOTAL_PRICE, STATUS) = (?, ?, ?, ?, ?::e_status_invoice) WHERE ID = ?";
+    static final String DELETE_QUERY = "DELETE FROM INVOICE WHERE ID = ?";
+
+    private InvoiceEntity parseRow(ResultSet resultSet) throws SQLException {
+        int id = resultSet.getInt("id");
+        int orderId = resultSet.getInt("order_id");
+        Date issueDate = resultSet.getDate("issue_date");
+        Date paymentDate = resultSet.getDate("payment_date");
+        BigDecimal totalPrice = resultSet.getBigDecimal("total_price");
+        String statusString = resultSet.getString("status").toUpperCase();
+        InvoiceStatus status = InvoiceStatus.valueOf(statusString);
+
+        return new InvoiceEntity(id, orderId, issueDate, paymentDate, totalPrice, status);
+    }
+
+    private void setInvoiceFieldsToStatement(InvoiceEntity invoice, PreparedStatement statement) throws SQLException {
+        statement.setInt(1, invoice.getOrderId());
+        statement.setDate(2, new java.sql.Date(invoice.getIssueDate().getTime()));
+        statement.setDate(3, new java.sql.Date(invoice.getPaymentDate().getTime()));
+        statement.setBigDecimal(4, invoice.getTotalPrice());
+        statement.setString(5, invoice.getStatus().name().toLowerCase());
+    }
+
+    @Override
+    public List<InvoiceEntity> getAll() {
+        List<InvoiceEntity> invoices = new ArrayList<>();
+        DBConnectorInterface dbConnector = DBConnectorPostgres.getInstance();
+        Connection connection = dbConnector.getConnection();
+        try {
+            Statement statement = connection.createStatement(ResultSet.TYPE_SCROLL_SENSITIVE,
+                    ResultSet.CONCUR_READ_ONLY);
+            ResultSet resultSet = statement.executeQuery(GET_ALL_QUERY);
+            resultSet.first();
+            do {
+                InvoiceEntity currentInvoice = parseRow(resultSet);
+                invoices.add(currentInvoice);
+            }
+            while (resultSet.next());
+            connection.close();
+            return invoices;
+        } catch (SQLException e) {
+            e.printStackTrace();
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public InvoiceEntity get(int id) {
+        InvoiceEntity invoice;
+        DBConnectorInterface dbConnector = DBConnectorPostgres.getInstance();
+        Connection connection = dbConnector.getConnection();
+        try {
+            PreparedStatement statement = connection.prepareStatement(GET_QUERY, ResultSet.TYPE_SCROLL_SENSITIVE,
+                    ResultSet.CONCUR_READ_ONLY);
+            statement.setInt(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            resultSet.first();
+            invoice = parseRow(resultSet);
+            connection.close();
+            return invoice;
+        } catch (SQLException e) {
+            e.printStackTrace();
+            throw new RuntimeException(e);
```

Comment:
Throw some kind of checked exception instead (e.g. it's very likely we'll want to know if there's no entity with such id and handle it gracefully)